group/announce.py

```diff
@@ -55,13 +55,16 @@ def add_announce(request, group_id):
         raise PermissionDenied
 
 @login_required
-def delete_announce(request, announce_id, group_id):
+def delete_announce(request, announce_id, group_id, redirect_page):
     group = get_group(group_id)
 
     if has_group_ownership(request.user, group) or has_group_coownership(request.user, group):  
         get_announce(announce_id).delete()
         logger.info('Announce: User %s delete Announce %s!' % (request.user.username, announce_id))
-        return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+        if redirect_page == 'detail':
```

Comment:
I prefer that
```
if viewall:
    # redirect to view all page
else:
    # redirect to detail page
```
The difference is that my code will handle unexpected cases.
it's equal to add this line ```else: redirect to detail```